gateway/utils/admin/admin.go

```diff
@@ -58,3 +58,20 @@ func (m *Manager) Login(user, pass string) (int, string, error) {
 
 	return http.StatusUnauthorized, "", errors.New("invalid credentials provided")
 }
+// RefreshToken is used to create a new token!
+func (m *Manager) RefreshToken(token string) (int,string,error){
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	// Parse the token to get userID and userRole
+	tokenClaims ,err := m.parseToken(token)
+	if err != nil {
+		return http.StatusUnauthorized,"", err
+	}
+	// Create a new token
+		newToken, err := m.createToken(map[string]interface{}{"id": tokenClaims["id"], "nodeID": tokenClaims["nodeID"]})
```

Comment:
if you carefully look at the login handler following claims are used to create token
![image](https://user-images.githubusercontent.com/24411676/73136915-f7734000-4078-11ea-92e1-a3bc4781519f.png)
"id" & "token" both have been assigned the same value that is "user"
so check if "id" key present in tokenClaims if yes then pass the same value for both keys of create token map that is "id" & "role"